src/Components/OATGraphViewer/Internal/OATGraphCustomNode.tsx

```diff
@@ -0,0 +1,383 @@
+    getGraphViewerIconStyles,
+    getGraphViewerActionButtonStyles
+} from '../OATGraphViewer.styles';
+import { ElementsContext } from './OATContext';
+import {
+    OATRelationshipHandleName,
+    OATComponentHandleName,
+    OATExtendHandleName,
+    OATUntargetedRelationshipName
+} from '../../../Models/Constants/Constants';
+import {
+    SET_OAT_PROPERTY_EDITOR_MODEL,
+    SET_OAT_DELETED_MODEL_ID,
+    SET_OAT_CONFIRM_DELETE_OPEN
+} from '../../../Models/Constants/ActionTypes';
+import {
+    getPropertyDisplayName,
+    isDisplayNameDefined
+} from '../../OATPropertyEditor/Utils';
+import OATTextFieldDisplayName from '../../../Pages/OATEditorPage/Internal/Components/OATTextFieldDisplayName';
+import OATTextFieldId from '../../../Pages/OATEditorPage/Internal/Components/OATTextFieldId';
+import IconRelationship from '../../../Resources/Static/relationshipTargeted.svg';
+import IconUntargeted from '../../../Resources/Static/relationshipUntargeted.svg';
+import IconInheritance from '../../../Resources/Static/relationshipInheritance.svg';
+import IconComponent from '../../../Resources/Static/relationshipComponent.svg';
+import Svg from 'react-inlinesvg';
+import { deepCopy } from '../../../Models/Services/Utils';
+
+const OATGraphCustomNode: React.FC<IOATGraphCustomNodeProps> = ({
+    data,
+    isConnectable
+}) => {
+    const { t } = useTranslation();
+    const [nameEditor, setNameEditor] = useState(false);
+    const [nameText, setNameText] = useState(getPropertyDisplayName(data));
+    const [idEditor, setIdEditor] = useState(false);
+    const [idText, setIdText] = useState(data.id);
+    const { dispatch, state, currentHovered, currentNodeIdRef } = useContext(
+        ElementsContext
+    );
+    const graphViewerStyles = getGraphViewerStyles();
+    const iconStyles = getGraphViewerIconStyles();
+    const actionButtonStyles = getGraphViewerActionButtonStyles();
+    const [handleHoverRelationship, setHandleHoverRelationship] = useState(
+        false
+    );
+    const [handleHoverComponent, setHandleHoverComponent] = useState(false);
+    const [handleHoverExtend, setHandleHoverExtend] = useState(false);
+    const [handleHoverUntargeted, setHandleHoverUntargeted] = useState(false);
+    const { model, models } = state;
+
+    const onNameClick = () => {
+        if (!state.modified) {
+            setNameText(getPropertyDisplayName(data));
+            setNameEditor(true);
+        }
+    };
+
+    const onIdClick = () => {
+        if (!state.modified) {
+            setIdEditor(true);
+        }
+    };
+
+    const onDelete = () => {
+        if (!state.modified) {
+            const dispatchDelete = () => {
+                dispatch({
+                    type: SET_OAT_DELETED_MODEL_ID,
+                    payload: data.id
+                });
+            };
+            dispatch({
+                type: SET_OAT_CONFIRM_DELETE_OPEN,
+                payload: { open: true, callback: dispatchDelete }
+            });
```

Comment:
This eventually triggers `onElementsRemove` on graph viewer. 
`onElementsRemove` does have and undo.
I'll keep this comment as a note in case we remove `deletedModelId`